context.go

```diff
@@ -101,19 +117,29 @@ func (d *DuplicateFieldError) Error() string {
 // this type context.  If you want to use fields from a different type context,
 // use TranslateTypeRecord.
 func (c *Context) LookupTypeRecord(fields []Field) (*TypeRecord, error) {
-	tv := tvPool.Get().(*[]byte)
-	*tv = AppendTypeValue((*tv)[:0], &TypeRecord{Fields: fields})
+	key := keyPool.Get().(*[]byte)
+	bytes := (*key)[:0]
+	for _, field := range fields {
+		bytes = binary.LittleEndian.AppendUint32(bytes, uint32(len(field.Name)))
+		bytes = append(bytes, []byte(field.Name)...)
```

Comment:
Nit: No need to convert string to []byte here.
```suggestion
		bytes = append(bytes, field.Name...)
```